src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.Metrics.cs

```diff
@@ -50,6 +51,13 @@ partial class DiagController
                     includeDefaults: true,
                     durationSeconds: durationSeconds);
 
+                // Allow sync I/O on livemetrics routes due to JsonCounterLogger's usage.
+                var syncIOFeature = HttpContext.Features.Get<IHttpBodyControlFeature>();
```

Comment:
Could you make this an extension method of HttpContext and Debug.Assert that the feature could be obtained?